CONTRIBUTING.md

```diff
@@ -0,0 +1,67 @@
+This is a community generated project and we appreciate contributions from other developers. If you would like to contribute, this document will help you.
+
+## Getting started
+
+First thing is to clone the repo and ensure you have everything running locally. Here are some steps to guide you:
+
+* To install the DNVM and DNX see [Getting Started with ASP.NET 5 and DNX](https://github.com/aspnet/Home/blob/dev/README.md)
+* Create a fork of this repository.
+* Clone you fork to your computer. Go to a command prompt and CD to your project folder. 
+* Ensure that you are running the latest unstable dev feed of the runtime by running the command `dnvm upgrade -u`.
+* Restore all packages by running the command `dnu restore`
+* At this point you should be able to open the solution in Visual Studio and run the sample project 
+    
+If you want to add a new provider, look at what providers are available for you to pick up in [the issues](https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers/issues). If you see one you want to work on, leave a comment stating that you are picking it up so we don't get people working on the same things. If you don't see the provider you want to work on listed, then just create an issue for it and then leave a comment that you’re implementing it.
+
+## Using the Yeoman generator
+
+To create a new provider, we have created a Yeoman generator to create the basic scaffolding for you. To install it please ensure you have completed the following:
+
+* Install NodeJS from [their website](https://nodejs.org/en/download/).
+* Install Yeoman by running `npm install -g yo`
+* Next install the generator by running `npm install -g generator-aspnet-oauth`
```

Comment:
this can be combined with the line above `npm install -g yo generator-aspnet-oauth`